File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/region/ImageComponentType.java
package com.enonic.wem.api.content.page.region;

public final class ImageComponentType
    extends ComponentType
{
    public final static ImageComponentType INSTANCE = new ImageComponentType();

    private static final ImageComponentDataSerializer dataSerializer = new ImageComponentDataSerializer();

    private ImageComponentType()
    {
        super( "image", ImageComponent.class );
    }

    @Override
    public ImageComponentDataSerializer getDataSerializer()
    {
        return dataSerializer;
    }
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/content/page/region/ImageComponentType.ts
module api.content.page.region {

    export class ImageComponentType extends ComponentType {

        private static INSTANCE: ImageComponentType = new ImageComponentType();

        constructor() {
            super("image");
        }

        newComponentBuilder(): ImageComponentBuilder {
            return new ImageComponentBuilder();
        }

        public static get(): ImageComponentType {
            return ImageComponentType.INSTANCE;
        }
    }

}
